web/packages/teleterm/src/services/fileStorage/fileStorage.ts

```diff
@@ -64,19 +76,26 @@ export function createFileStorage(opts: {
     }
   }
 
-  function get<T>(key?: string): T {
-    return key ? state[key] : (state as T);
-  }
-
   function replace(json: any): void {
+    if (discardUpdates) {
+      return;
+    }
     state = json;
     stringifyAndWrite();
   }
 
+  function get<T>(key?: string): T {
+    return key ? state[key] : (state as T);
+  }
```

Comment:
Absolutely agree, `get` in its current form doesn't make any sense, it only makes casting a little bit easier. 
I applied your patch, it is still better than what we had. 

>well, it'd be just as unsound as anything else as we don't verify that the contents of the file actually have the shape we expect, this could be done with zod

Parsing the file with zod sounds nice, although we would have to think how to combine it with the `ConfigService` so the file wouldn't be validated twice.

>But instead of setting it on get, we'd need to update createFileStorage to accept a generic type State which could be then used in other methods.

👍

Reply:
I added a comment to `get` which links to this discussion.